ui/src/components/issuer-state/IssuerState.tsx

```diff
@@ -0,0 +1,266 @@
+import {
+  Avatar,
+  Button,
+  Card,
+  Divider,
+  Row,
+  Space,
+  Table,
+  Tag,
+  Tooltip,
+  Typography,
+  message,
+} from "antd";
+import { ColumnsType } from "antd/es/table";
+import dayjs from "dayjs";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { Link } from "react-router-dom";
+
+import { APIError } from "src/adapters/api";
+import { getTransactions, publishState, retryPublishState } from "src/adapters/api/issuer-state";
+import { ReactComponent as IconSwitch } from "src/assets/icons/switch-horizontal.svg";
+import { ErrorResult } from "src/components/shared/ErrorResult";
+import { SiderLayoutContent } from "src/components/shared/SiderLayoutContent";
+import { TableCard } from "src/components/shared/TableCard";
+import { useEnvContext } from "src/contexts/Env";
+import { useIssuerStateContext } from "src/contexts/IssuerState";
+import { Transaction } from "src/domain";
+import { AsyncTask, isAsyncTaskDataAvailable, isAsyncTaskStarting } from "src/utils/async";
+import { makeRequestAbortable } from "src/utils/browser";
+
+import { ISSUER_STATE, POLLING_INTERVAL, STATUS } from "src/utils/constants";
+import { formatDate } from "src/utils/forms";
+
+export function IssuerState() {
+  const env = useEnvContext();
+  const { refreshStatus, status } = useIssuerStateContext();
+
+  const [isPublishing, setIsPublishing] = useState<boolean>(false);
+  const [isRetrying, setIsRetrying] = useState<boolean>(false);
+  const [transactions, setTransactions] = useState<AsyncTask<Transaction[], APIError>>({
+    status: "pending",
+  });
+
+  const transactionsList = useMemo(
+    () => (isAsyncTaskDataAvailable(transactions) ? transactions.data : []),
+    [transactions]
+  );
+
+  const failedTransaction = transactionsList.find((transaction) => transaction.status === "failed");
+  const disablePublishState =
+    failedTransaction !== undefined || !isAsyncTaskDataAvailable(status) || !status.data;
+
+  const publish = () => {
+    setIsPublishing(true);
+    void publishState({ env }).then((response) => {
+      if (response.isSuccessful) {
+        void message.success("Issuer state is being published");
+      } else {
+        void message.error(response.error.message);
+      }
+
+      void refreshStatus();
+      void fetchTransactions();
+
+      setIsPublishing(false);
+    });
+  };
+
+  const retry = () => {
```

Comment:
This is almost identical to `publish` ... can we DRY this up? Maybe at least with the success message?